orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/instance/DeregisterInstancesFromLoadBalancerStage.groovy

```diff
@@ -21,31 +21,24 @@ import com.netflix.spinnaker.orca.clouddriver.OortService
 import com.netflix.spinnaker.orca.clouddriver.tasks.MonitorKatoTask
 import com.netflix.spinnaker.orca.clouddriver.tasks.instance.DeregisterInstancesFromLoadBalancerTask
 import com.netflix.spinnaker.orca.clouddriver.tasks.instance.WaitForDownInstanceHealthTask
-import com.netflix.spinnaker.orca.pipeline.LinearStage
-import com.netflix.spinnaker.orca.pipeline.model.Stage
+import com.netflix.spinnaker.orca.pipeline.StageDefinitionBuilder
 import groovy.transform.CompileStatic
-import org.springframework.batch.core.Step
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Component
 
 @Component
 @CompileStatic
-class DeregisterInstancesFromLoadBalancerStage extends LinearStage {
-  public static final String PIPELINE_CONFIG_TYPE = "deregisterInstancesFromLoadBalancer"
-
+class DeregisterInstancesFromLoadBalancerStage implements StageDefinitionBuilder {
   @Autowired
   OortService oortService
 
-  DeregisterInstancesFromLoadBalancerStage() {
-    super(PIPELINE_CONFIG_TYPE)
-  }
-
   @Override
-  public List<Step> buildSteps(Stage stage) {
-    def step1 = buildStep(stage, "deregisterInstances", DeregisterInstancesFromLoadBalancerTask)
-    def step2 = buildStep(stage, "monitorInstances", MonitorKatoTask)
-    def step3 = buildStep(stage, "waitForLoadBalancerState", WaitForDownInstanceHealthTask)
-    [step1, step2, step3]
+  List<StageDefinitionBuilder.TaskDefinition> taskGraph() {
+    return [
+      new StageDefinitionBuilder.TaskDefinition("deregisterInstances", DeregisterInstancesFromLoadBalancerTask),
```

Comment:
We should move that class up to the package level since IntelliJ doesn't auto-static import